frontend/src/pages/projects/screens/detail/connections/ManageConnectionsModal.tsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import { Modal } from '@patternfly/react-core';
+import DashboardModalFooter from '~/concepts/dashboard/DashboardModalFooter';
+import ConnectionTypePreview from '~/concepts/connectionTypes/ConnectionTypePreview';
+import {
+  Connection,
+  ConnectionTypeConfigMapObj,
+  ConnectionTypeFieldType,
+  ConnectionTypeValueType,
+  isConnectionTypeDataField,
+} from '~/concepts/connectionTypes/types';
+import { translateDisplayNameForK8s } from '~/concepts/k8s/utils';
+import { ProjectKind, SecretKind } from '~/k8sTypes';
+import { useK8sNameDescriptionFieldData } from '~/concepts/k8s/K8sNameDescriptionField/K8sNameDescriptionField';
+import { K8sNameDescriptionFieldData } from '~/concepts/k8s/K8sNameDescriptionField/types';
+
+type Props = {
+  connection?: Connection;
+  connectionTypes?: ConnectionTypeConfigMapObj[];
+  project?: ProjectKind;
```

Comment:
These should all be required fields.

Reply:
I think `connection?: Connection` can remain optional, since this is the scenario for a new connection. If it's edit, then `connection` will be a filled out value.

I can update the other 2 though for sure